packages/typescript-estree/src/convert.ts

```diff
@@ -119,7 +119,12 @@ export class Converter {
 
     if (result && this.options.shouldProvideParserServices) {
       this.tsNodeToESTreeNodeMap.set(node, result);
-      this.esTreeNodeToTSNodeMap.set(result, node);
+      if (node.kind !== SyntaxKind.ParenthesizedExpression) {
```

Comment:
this is really nice catch, but we should extend it also to include `SyntaxKind.ComputedPropertyName`

its same issue

Reply:
```ts
class foo {
	['test']: string
}
```

Reply:
@JamesHenry if you think this should be addressed in separated pr, i'm ok with that

Reply:
Added the `ComputedPropertyName` case.